worlds/Files.py

```diff
@@ -135,31 +126,61 @@ def read(self, file: Optional[Union[str, BinaryIO]] = None) -> None:
                         message = f"{arg0} - "
                 raise InvalidDataError(f"{message}This might be the incorrect world version for this file") from e
 
-    def read_contents(self, opened_zipfile: zipfile.ZipFile) -> None:
+    def read_contents(self, opened_zipfile: zipfile.ZipFile) -> Dict[str, Any]:
         with opened_zipfile.open("archipelago.json", "r") as f:
             manifest = json.load(f)
         if manifest["compatible_version"] > self.version:
             raise Exception(f"File (version: {manifest['compatible_version']}) too new "
                             f"for this handler (version: {self.version})")
+        return manifest
+
+    def get_manifest(self) -> Dict[str, Any]:
+        return {
+            # minimum version of patch system expected for patching to be successful
+            "compatible_version": 5,
+            "version": container_version,
+        }
+
+
+class APPlayerContainer(APContainer):
+    """A zipfile containing at least archipelago.json meant for a player"""
+    game: ClassVar[Optional[str]] = None
+
+    player: Optional[int]
+    player_name: str
+    server: str
+
+    def __init__(self, path: Optional[str] = None, player: Optional[int] = None,
+                 player_name: str = "", server: str = ""):
+        super().__init__(path)
+        self.player = player
+        self.player_name = player_name
+        self.server = server
+
+    def read_contents(self, opened_zipfile: zipfile.ZipFile) -> Dict[str, Any]:
+        manifest = super().read_contents(opened_zipfile)
         self.player = manifest["player"]
         self.server = manifest["server"]
         self.player_name = manifest["player_name"]
+        return manifest
 
     def get_manifest(self) -> Dict[str, Any]:
-        return {
+        manifest = super().get_manifest()
+        manifest.update({
             "server": self.server,  # allow immediate connection to server in multiworld. Empty string otherwise
             "player": self.player,
             "player_name": self.player_name,
             "game": self.game,
             # minimum version of patch system expected for patching to be successful
             "compatible_version": 5,
             "version": container_version,
```

Comment:
Any reason to keep the `compatible_version` and `version` here? They are already added to the manifest in `APContainer`.